docs/documentation/siddhi-4.0.md

```diff
@@ -2345,4 +2352,167 @@ You can use Store extension type to work with data/events stored in various data
                           
           eg:- Define value for property 'storeType': RDBMS
     
-* Finally confirm all property values are correct or not by typing Y or press Enter, else type N
\ No newline at end of file
+* Finally confirm all property values are correct or not by typing Y or press Enter, else type N.
+
+## Configuring and Monitoring Siddhi Applications
+
+This section explains how to use the `@app` annotation to generate statistics for Siddhi applications as well as improve the performance of Siddhi applications.
+
+### @app:statistics
+
+To evaluate the performance of an application, you can enable the statistics of a Siddhi application to be published. This is done via the `@app:statistics` annotation that can be added to a Siddhi application as shown in the following example.
+
+```sql
+@app:statistics(reporter = 'console')
+```
+The following elements are configured with this annotation.
+
+|Annotation| Description| Default Value|
+| ------------- |-------------|-------------|
+|`reporter`|The interface in which statistics for the Siddhi application are published. Possible values are as follows:<br/> `console`<br/> `jmx`<br/> `jdbc`<br/> |`console`|
+|`interval`|The time interval (in seconds) at  which the statistics for the Siddhi application are reported.|`60`|
+|`include`| This allows you to specify the types of metrics to be included in the reporting as a comma-separated list. It is also possible to use wild cards.| All (*.*)|
+
+The metrics are reported in the following format.
+`org.wso2.siddhi.SiddhiApps.<SiddhiAppName>.Siddhi.<Component Type>.<Component Name>. <Metrics name>`
+
+The following table lists the types of metrics supported for different Siddhi application component types.
+
+|Component Type|Metrics Type|
+| ------------- |-------------|
+|Stream|Throughput<br/>The size of the buffer if parallel processing is enabled via the @async annotation.|
+|Trigger|Throughput (Trigger and Stream)|
+|Source|Throughput|
+|Sink|Throughput|
+|Mapper|Latency<br/>Input/output throughput<br/>
+|Table|Memory<br/>Throughput (For all operations)<br/>Throughput (For all operations)|
+|Query|Memory<br/>Latency|
+|Window|Throughput (For all operations)<br/>Latency (For all operation)|
+|Partition|Throughput (For all operations)<br/>Latency (For all operation)|
+
+
+
+e.g., the following is a Siddhi application that includes the `@app` annotation to report performance statistics.
+
+```sql
+@App:name('TestMetrics')
```

Comment:
Shall we add a simplified Siddhi app, with all statistics here, it will be easy to correlate here